flixel/addons/editors/tiled/TiledMap.hx

```diff
@@ -135,73 +110,40 @@ class TiledMap
 			}
 		}
 		
-		// load layer
-		for (node in source.nodes.layer)
+		// Load tile and object layers
+		for (el in source.elements)
 		{
-			name = node.att.name;
-			
-			if (!noLoadHash.exists(name))
+			if (noLoadHash.exists(el.att.name)) continue;
+			if (el.name.toLowerCase() == "layer")
 			{
-				layers.push(new TiledLayer(node, this));
+				layers.push(new TiledTileLayer(el, this));
 			}
-		}
-		
-		// load object group
-		for (node in source.nodes.objectgroup)
-		{
-			name = node.att.name;
-			
-			if (!noLoadHash.exists(name))
+			else if (el.name.toLowerCase() == "objectgroup")
 			{
-				objectGroups.push(new TiledObjectGroup(node, this));
+				layers.push(new TiledObjectLayer(el, this));
 			}
 		}
 	}
 	
-	public function getTileSet(Name:String):TiledTileSet
+	public function getTileSet(name:String):TiledTileSet
 	{
-		return tilesets.get(Name);
-	}
-	
-	public function getLayer(Name:String):TiledLayer
-	{
-		var i = layers.length;
-		
-		while (i > 0)
-		{
-			if (layers[--i].name == Name)
-			{
-				return layers[i];
-			}
-		}
-		
-		return null;
+		return tilesets.get(name);
 	}
 	
-	public function getObjectGroup(Name:String):TiledObjectGroup
-	{
-		var i = objectGroups.length;
-		
-		while (i > 0)
-		{
-			if (objectGroups[--i].name == Name)
-			{
-				return objectGroups[i];
-			}
-		}
-		
-		return null;
+	public function getLayer(name:String):TiledLayer
+	{	
+		return layerMap.get(name);
 	}
 	
 	// works only after TiledTileSet has been initialized with an image...
```

Comment:
If you're already at it, you could change this to be a `/* */`-style comment. (FD ignores `//`-comments in completion).